src/popup/components/Settings/WalletInformation.tsx

```diff
@@ -0,0 +1,70 @@
+import React from 'react';
+import { useForm } from 'react-hook-form';
+import { ArrowBack } from '@/popup/components/Icons';
+import { Input } from '@/popup/components/ui/Input';
+import { Label } from '@/popup/components/ui/Label';
+import { Code } from '@/popup/components/ui/Code';
+import { Button } from '@/popup/components/ui/Button';
+import { useMessage } from '@/popup/lib/context';
+import type { PopupStore } from '@/shared/types';
+
+interface WalletInformationProps {
+  publicKey: PopupStore['publicKey'];
+  walletAddress: PopupStore['walletAddress'];
+}
+
+export const WalletInformation = ({
+  publicKey,
+  walletAddress,
+}: WalletInformationProps) => {
+  const message = useMessage();
+  const {
+    handleSubmit,
+    formState: { isSubmitting },
+  } = useForm();
+
+  return (
+    <div className="flex h-full flex-col gap-8">
+      {/* TODO: Improve error handling */}
+      <form
+        className="space-y-4"
+        onSubmit={handleSubmit(async () => {
+          await message.send('DISCONNECT_WALLET');
+          window.location.reload();
```

Comment:
We should check the response here. But I guess it's going to be part of the above TODO item, right?